src/oidc_clients/api/clients.ts

```diff
@@ -0,0 +1,70 @@
+import { get, getAllPages } from 'common/utils/api';
+import { IAuthUser } from '../../authentication/models/User';
+import { deleteR } from '../../common/utils/api';
+import { IOidcClient } from '../models/client';
+import { IResponseType } from '../models/responsetype';
+const API_BASE = '/api/v1/oidc/';
+
+// api for geting oidc client information and for creating new oidc clients
+type ClientResponseType = { response_types: number[] } & Omit<IOidcClient, 'response_types'>;
+
+// fetches specific client
+export const getClient = async (id: number): Promise<IOidcClient> => {
+  const client = await get<ClientResponseType>(API_BASE + 'clients/' + id);
+  const responseTypes = await getResponseTypes();
+  const newResponseTypes = client.response_types.flatMap((response_type) => {
+    const responseType = responseTypes.get(response_type);
+    return responseType ? [responseType] : [];
+  });
+
+  const newClient: IOidcClient = { ...client, response_types: newResponseTypes };
+  return newClient;
+};
+
+// fetches all clients
+export const getAllClients = async (): Promise<IOidcClient[]> => {
+  const clients = await getAllPages<ClientResponseType>(API_BASE + 'clients/');
+  const responseTypes = await getResponseTypes();
+  const newClients: IOidcClient[] = [];
+
+  for (const client of clients) {
+    const newResponseTypes = client.response_types.flatMap((response_type) => {
+      const responseType = responseTypes.get(response_type);
+      return responseType ? [responseType] : [];
+    });
+
+    const newClient: IOidcClient = { ...client, response_types: newResponseTypes };
+    newClients.push(newClient);
+  }
+
+  return newClients;
+};
+
+export const getResponseTypes = async (): Promise<Map<number, IResponseType>> => {
+  const responseTypes = await getAllPages<IResponseType>(API_BASE + 'response-types/');
+  const responseTypeMap: Map<number, IResponseType> = new Map();
+
+  for (const responseType of responseTypes) {
+    responseTypeMap.set(responseType.id, responseType);
+  }
+
+  return responseTypeMap;
+};
+
+// fetch clients owned by me
+// todo: filter matches on username, client.owner should probably contain user id aswell
+export const getMyClients = async (user: IAuthUser): Promise<IOidcClient[]> => {
+  if (!user) {
+    return [];
+  }
+  const clients = await getAllClients();
+  const filtered = clients.filter((client) => {
+    return client.owner && client.owner.username === user.profile.preferred_username;
+  });
```

Comment:
Could we filter this in the query to the API instead?

Reply:
How would I approach that? I have tested using filter "name=nibble-cli" and "owner.username=jannetr" but it doesn't actually filter out anything.

Reply:
We would have to add it in the back-end, could look into it for future stuff